input/script_utils.go

```diff
@@ -1323,6 +1323,54 @@ func CommitScriptToRemoteConfirmed(key *btcec.PublicKey) ([]byte, error) {
 	return builder.Script()
 }
 
+// TaprootCommitScriptToRemote constructs a taproot witness program for the
+// output on the commitment transaction for the remote party. For the top level
+// key spend, we'll use the combined funding key (musig2.KeyAgg(k1, k2)), as a
+// sort of practical NUMs point (the local party would never sign for this). We
+// then commit to a single tapscript leaf that holds the normal CSV 1 delay
+// script.
+//
+// Our single tapleaf will use the following script:
+//
+//	<remotepubkey> OP_CHECKSIG
+//	OP_CHECKSEQUENCEVERIFY
+//
+// The CSV clause is a bit subtle, but OP_CHECKSIG will return true if it
+// succeeds, which then enforces our 1 CSV. The true will remain on the stack,
```

Comment:
in Bitcoin Core the check is here: https://github.com/bitcoin/bitcoin/blob/master/src/script/interpreter.cpp#L1819-L1825

in the selected lines the script is executed (line 1820), it must succeed to continue (it will return false on failure), if the stack size is not 1 (line 1823), false is returned, and finally the top stack element has to be true as defined by `CastToBool` (for example 0 is false and 1 is true) (line 1824). Only if it passed these if statements it will get to the `return true` at the end (set_error returns false)